tfs/src/main/java/hudson/plugins/tfs/TeamPushTrigger.java

```diff
@@ -76,6 +76,14 @@ public void setJobContext(final String jobContext) {
         this.jobContext = jobContext;
     }
 
+    public Job<?, ?> getJob() {
+        return this.job;
+    }
+
+    public void setJob(final Job<?, ?> job) {

```

Comment:
It seems wrong to have a setter for Job in this class. I don't see it being used any way.

Reply:
I see it being used now. It still seems strange that you can change the Job that the trigger was created with.

Reply:
You are right, just need to be cautions job isn't null when TeamPushTrigger is initialized. I checked all the callers, and it's the case, so both setters and getters of jobs are unnecessary for now. Removed them.